tensorflow/lite/kernels/eigen_support.cc

```diff
@@ -103,7 +103,7 @@ class LazyEigenThreadPoolHolder {
   // Updates the thread count, invalidating the ThreadPoolDevice if necessary.
   void SetNumThreads(int num_threads) {
     const int target_num_threads =
-        num_threads != -1 ? num_threads : kDefaultNumThreadpoolThreads;
+        num_threads > -1 ? num_threads : kDefaultNumThreadpoolThreads;
```

Comment:
What happens if you set num threads to 0? If we're fixing this path, we should cover all the edge cases.

Reply:
num thread 0 has special meaning in Eigen, so i had included it. Also we need to make sure there is no negative value set into it. Current code has bug, if by any chance negative value like -2 is set, then it will passed to Eigen, which is not right, so the change.